src/sonic-config-engine/sonic_device_util.py

```diff
@@ -125,6 +133,78 @@ def get_all_namespaces():
 
     return {'front_ns':front_ns, 'back_ns':back_ns}
 
+def get_port_config_from_all_asics(): 
+    if not is_multi_npu():
+        return get_port_config()
+
+    platform = get_platform()
+    hwsku = get_hwsku()
+    all_ports = {}
+    ns_list  = get_namespaces()
+    
+    for ns in ns_list:
+        asic_id = get_npu_id_from_name(ns)
+        (ports, _, _) = get_port_config(hwsku=hwsku, platform=platform, asic=asic_id)
+        all_ports.update(ports)
+    
+    return all_ports
+    
+def get_port_role(port_name):
+    ports_config = get_port_config_from_all_asics()
+    if not ports_config[port_name].has_key('role'):
+        return EXTERNAL_PORT
+    
+    role = ports_config[port_name]['role']
+    return role
+
+def is_port_internal(port_name):
+    role = get_port_role(port_name)
+    if role is INTERNAL_PORT:
+        return True
+    return False
+
+def is_port_channel_internal(port_channel, namespace=None):
+    if not is_multi_npu():
+        return False
+    if namespace is None:
+        ns_list  = get_namespaces()
+    else:
+        ns_list = [namespace]
```

Comment:
check of valid namepsace it not already checked. Raise exception if invalid namespace.

Reply:
IMO, the namespace validation should be done by the caller ( CLI handler)